[2025-01-03 17:03:48,399] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) is unavailable or invalid due to cause: session timed out without receiving a heartbeat response. isDisconnected: false. Rediscovery will be attempted. (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:1000)
[2025-01-03 17:03:48,402] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Requesting disconnect from last known coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:1013)
[2025-01-03 17:03:48,402] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Client requested disconnect from node 2147483647 (org.apache.kafka.clients.NetworkClient:344)
[2025-01-03 17:03:48,404] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Discovered group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:937)
[2025-01-03 17:03:48,448] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) is unavailable or invalid due to cause: coordinator unavailable. isDisconnected: false. Rediscovery will be attempted. (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:1000)
[2025-01-03 17:03:48,449] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Requesting disconnect from last known coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:1013)
[2025-01-03 17:03:48,503] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) is unavailable or invalid due to cause: session timed out without receiving a heartbeat response. isDisconnected: false. Rediscovery will be attempted. (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:1000)
[2025-01-03 17:03:48,503] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Requesting disconnect from last known coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:1013)
[2025-01-03 17:03:48,504] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Client requested disconnect from node 2147483647 (org.apache.kafka.clients.NetworkClient:344)
[2025-01-03 17:03:48,507] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Discovered group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:937)
[2025-01-03 17:03:48,508] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) is unavailable or invalid due to cause: coordinator unavailable. isDisconnected: false. Rediscovery will be attempted. (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:1000)
[2025-01-03 17:03:48,508] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Requesting disconnect from last known coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:1013)
[2025-01-03 17:03:48,570] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Discovered group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:937)
[2025-01-03 17:03:48,618] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Discovered group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:937)
[2025-01-03 17:03:59,546] INFO [mongodb-sink|task-0] Exception in monitor thread while connecting to server ist-shared-shard-00-01.n0kts.mongodb.net:27017 (org.mongodb.driver.cluster:76)
com.mongodb.MongoSocketReadTimeoutException: Timeout while receiving message
	at com.mongodb.internal.connection.InternalStreamConnection.translateReadException(InternalStreamConnection.java:696)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:574)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveCommandMessageResponse(InternalStreamConnection.java:410)
	at com.mongodb.internal.connection.InternalStreamConnection.receive(InternalStreamConnection.java:369)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.lookupServerDescription(DefaultServerMonitor.java:221)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.run(DefaultServerMonitor.java:157)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:278)
	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:304)
	at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:346)
	at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:796)
	at java.base/java.net.Socket$SocketInputStream.implRead(Socket.java:1116)
	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:1103)
	at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:489)
	at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:483)
	at java.base/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:70)
	at java.base/sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1462)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1068)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:110)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:135)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveResponseBuffers(InternalStreamConnection.java:713)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:571)
	... 5 more
[2025-01-03 17:04:00,564] INFO [mariadb-connector|task-0] Stopped reading binlog after 169 events, last recorded offset: {ts_sec=1735940249, file=mariadb-bin.000001, pos=88942, server_id=1, event=1} (io.debezium.connector.binlog.BinlogStreamingChangeEventSource:1218)
[2025-01-03 17:04:00,564] INFO [mariadb-connector|task-0] Creating thread debezium-mysqlconnector-kafka-binlog-client (io.debezium.util.Threads:287)
[2025-01-03 17:04:00,598] INFO [mariadb-connector|task-0] Connected to binlog at localhost:3306, starting at BinlogOffsetContext{sourceInfoSchema=Schema{io.debezium.connector.mysql.Source:STRUCT}, sourceInfo=BinlogSourceInfo{currentGtid='null', currentBinlogFilename='mariadb-bin.000001', currentBinlogPosition=88901, currentRowNumber=0, serverId=1, sourceTime=2025-01-03T21:37:29Z, threadId=-1, currentQuery='null', tableIds=[leafy_factory.product_cost], databaseName='leafy_factory'}, snapshotCompleted=true, transactionContext=TransactionContext [currentTransactionId=null, perTableEventCount={}, totalEventCount=0], restartGtidSet='null', currentGtidSet='null', restartBinlogFilename='mariadb-bin.000001', restartBinlogPosition=88942, restartRowsToSkip=0, restartEventsToSkip=1, currentEventLengthInBytes=41, inTransaction=false, transactionId='null', incrementalSnapshotContext=IncrementalSnapshotContext [windowOpened=false, chunkEndPosition=null, dataCollectionsToSnapshot=[], lastEventKeySent=null, maximumKey=null]} (io.debezium.connector.binlog.BinlogStreamingChangeEventSource:1232)
[2025-01-03 17:04:00,744] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:28, serverValue:1322989}] to ist-shared-shard-00-01.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:04:00,746] INFO [mongodb-sink|task-0] Monitor thread successfully connected to server with description ServerDescription{address=ist-shared-shard-00-01.n0kts.mongodb.net:27017, type=REPLICA_SET_PRIMARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=25, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=600295834, setName='atlas-133bp7-shard-0', canonicalAddress=ist-shared-shard-00-01.n0kts.mongodb.net:27017, hosts=[ist-shared-shard-00-02.n0kts.mongodb.net:27017, ist-shared-shard-00-01.n0kts.mongodb.net:27017, ist-shared-shard-00-00.n0kts.mongodb.net:27017], passives=[], arbiters=[], primary='ist-shared-shard-00-01.n0kts.mongodb.net:27017', tagSet=TagSet{[Tag{name='availabilityZone', value='cac1-az2'}, Tag{name='diskState', value='READY'}, Tag{name='nodeType', value='ELECTABLE'}, Tag{name='provider', value='AWS'}, Tag{name='region', value='CA_CENTRAL_1'}, Tag{name='workloadType', value='OPERATIONAL'}]}, electionId=7fffffff000000000000010a, setVersion=71, topologyVersion=TopologyVersion{processId=67694b4e6118cd58a4791747, counter=41}, lastWriteDate=Fri Jan 03 17:04:00 CST 2025, lastUpdateTimeNanos=66473332074750} (org.mongodb.driver.cluster:71)
[2025-01-03 17:04:00,746] INFO [mongodb-sink|task-0] Discovered replica set primary ist-shared-shard-00-01.n0kts.mongodb.net:27017 (org.mongodb.driver.cluster:71)
[2025-01-03 17:04:01,547] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:29, serverValue:1322993}] to ist-shared-shard-00-01.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:04:03,446] INFO [mongodb-sink|task-0] Exception in monitor thread while connecting to server ist-shared-shard-00-00.n0kts.mongodb.net:27017 (org.mongodb.driver.cluster:76)
com.mongodb.MongoSocketReadTimeoutException: Timeout while receiving message
	at com.mongodb.internal.connection.InternalStreamConnection.translateReadException(InternalStreamConnection.java:696)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:574)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveCommandMessageResponse(InternalStreamConnection.java:410)
	at com.mongodb.internal.connection.InternalStreamConnection.receive(InternalStreamConnection.java:369)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.lookupServerDescription(DefaultServerMonitor.java:221)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.run(DefaultServerMonitor.java:157)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:278)
	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:304)
	at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:346)
	at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:796)
	at java.base/java.net.Socket$SocketInputStream.implRead(Socket.java:1116)
	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:1103)
	at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:489)
	at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:483)
	at java.base/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:70)
	at java.base/sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1462)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1068)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:110)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:135)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveResponseBuffers(InternalStreamConnection.java:713)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:571)
	... 5 more
[2025-01-03 17:04:04,045] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:30, serverValue:1172582}] to ist-shared-shard-00-00.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:04:04,045] INFO [mongodb-sink|task-0] Monitor thread successfully connected to server with description ServerDescription{address=ist-shared-shard-00-00.n0kts.mongodb.net:27017, type=REPLICA_SET_SECONDARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=25, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=390459667, setName='atlas-133bp7-shard-0', canonicalAddress=ist-shared-shard-00-00.n0kts.mongodb.net:27017, hosts=[ist-shared-shard-00-02.n0kts.mongodb.net:27017, ist-shared-shard-00-01.n0kts.mongodb.net:27017, ist-shared-shard-00-00.n0kts.mongodb.net:27017], passives=[], arbiters=[], primary='ist-shared-shard-00-01.n0kts.mongodb.net:27017', tagSet=TagSet{[Tag{name='availabilityZone', value='cac1-az1'}, Tag{name='diskState', value='READY'}, Tag{name='nodeType', value='ELECTABLE'}, Tag{name='provider', value='AWS'}, Tag{name='region', value='CA_CENTRAL_1'}, Tag{name='workloadType', value='OPERATIONAL'}]}, electionId=null, setVersion=71, topologyVersion=TopologyVersion{processId=675b24d1d1431a587be694a8, counter=19}, lastWriteDate=Fri Jan 03 17:04:03 CST 2025, lastUpdateTimeNanos=66476632888000} (org.mongodb.driver.cluster:71)
[2025-01-03 17:04:05,135] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:31, serverValue:1172584}] to ist-shared-shard-00-00.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:04:05,508] INFO [mongodb-sink|task-0] Exception in monitor thread while connecting to server ist-shared-shard-00-02.n0kts.mongodb.net:27017 (org.mongodb.driver.cluster:76)
com.mongodb.MongoSocketReadTimeoutException: Timeout while receiving message
	at com.mongodb.internal.connection.InternalStreamConnection.translateReadException(InternalStreamConnection.java:696)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:574)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveCommandMessageResponse(InternalStreamConnection.java:410)
	at com.mongodb.internal.connection.InternalStreamConnection.receive(InternalStreamConnection.java:369)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.lookupServerDescription(DefaultServerMonitor.java:221)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.run(DefaultServerMonitor.java:157)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:278)
	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:304)
	at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:346)
	at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:796)
	at java.base/java.net.Socket$SocketInputStream.implRead(Socket.java:1116)
	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:1103)
	at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:489)
	at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:483)
	at java.base/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:70)
	at java.base/sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1462)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1068)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:110)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:135)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveResponseBuffers(InternalStreamConnection.java:713)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:571)
	... 5 more
[2025-01-03 17:04:05,950] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:32, serverValue:714217}] to ist-shared-shard-00-02.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:04:05,951] INFO [mongodb-sink|task-0] Monitor thread successfully connected to server with description ServerDescription{address=ist-shared-shard-00-02.n0kts.mongodb.net:27017, type=REPLICA_SET_SECONDARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=25, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=339668250, setName='atlas-133bp7-shard-0', canonicalAddress=ist-shared-shard-00-02.n0kts.mongodb.net:27017, hosts=[ist-shared-shard-00-02.n0kts.mongodb.net:27017, ist-shared-shard-00-01.n0kts.mongodb.net:27017, ist-shared-shard-00-00.n0kts.mongodb.net:27017], passives=[], arbiters=[], primary='ist-shared-shard-00-01.n0kts.mongodb.net:27017', tagSet=TagSet{[Tag{name='availabilityZone', value='cac1-az4'}, Tag{name='diskState', value='READY'}, Tag{name='nodeType', value='ELECTABLE'}, Tag{name='provider', value='AWS'}, Tag{name='region', value='CA_CENTRAL_1'}, Tag{name='workloadType', value='OPERATIONAL'}]}, electionId=null, setVersion=71, topologyVersion=TopologyVersion{processId=67694a2575e55e11df78dcd1, counter=14}, lastWriteDate=Fri Jan 03 17:04:05 CST 2025, lastUpdateTimeNanos=66478538101416} (org.mongodb.driver.cluster:71)
[2025-01-03 17:04:06,278] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:33, serverValue:714218}] to ist-shared-shard-00-02.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:20:16,720] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) is unavailable or invalid due to cause: session timed out without receiving a heartbeat response. isDisconnected: false. Rediscovery will be attempted. (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:1000)
[2025-01-03 17:20:16,720] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Disconnecting from node 0 due to request timeout. (org.apache.kafka.clients.NetworkClient:871)
[2025-01-03 17:20:16,720] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Requesting disconnect from last known coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:1013)
[2025-01-03 17:20:16,720] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Client requested disconnect from node 2147483647 (org.apache.kafka.clients.NetworkClient:344)
[2025-01-03 17:20:16,720] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Cancelled in-flight FETCH request with correlation id 26698 due to node 0 being disconnected (elapsed time since creation: 925551ms, elapsed time since send: 925551ms, throttle time: 0ms, request timeout: 30000ms) (org.apache.kafka.clients.NetworkClient:364)
[2025-01-03 17:20:16,766] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) is unavailable or invalid due to cause: session timed out without receiving a heartbeat response. isDisconnected: false. Rediscovery will be attempted. (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:1000)
[2025-01-03 17:20:16,766] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Requesting disconnect from last known coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:1013)
[2025-01-03 17:20:16,766] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Error sending fetch request (sessionId=844381896, epoch=22770) to node 0: (org.apache.kafka.clients.FetchSessionHandler:618)
org.apache.kafka.common.errors.DisconnectException
[2025-01-03 17:20:16,776] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Client requested disconnect from node 2147483647 (org.apache.kafka.clients.NetworkClient:344)
[2025-01-03 17:20:16,769] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Discovered group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:937)
[2025-01-03 17:20:16,777] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) is unavailable or invalid due to cause: coordinator unavailable. isDisconnected: false. Rediscovery will be attempted. (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:1000)
[2025-01-03 17:20:16,795] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Requesting disconnect from last known coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:1013)
[2025-01-03 17:20:16,850] INFO [mongodb-sink|task-0] [Consumer clientId=connector-consumer-mongodb-sink-0, groupId=connect-mongodb-sink] Discovered group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:937)
[2025-01-03 17:20:16,878] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Discovered group coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:937)
[2025-01-03 17:20:16,879] WARN [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] worker poll timeout has expired. The last known action being performed by the worker is : not known and may contribute to the timeout. Please review the last action (if known) for any corrective actions. One of the ways of addressing this can be increasing the rebalance.timeout.ms configuration value. Please note that rebalance.timeout.ms also controls the maximum allowed time for each worker to join the group once a rebalance has begun so the set value should not be very high (org.apache.kafka.connect.runtime.distributed.DistributedHerder:2764)
[2025-01-03 17:20:16,880] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Member connect-192.168.1.5:8083-232673f8-0ea4-4755-83b5-880c797c9fb6 sending LeaveGroup request to coordinator 192.168.1.5:9092 (id: 2147483647 rack: null) due to worker poll timeout has expired. (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:1174)
[2025-01-03 17:20:16,881] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Resetting generation and member id due to: consumer pro-actively leaving the group (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:1056)
[2025-01-03 17:20:16,881] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Rebalance started (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:243)
[2025-01-03 17:20:16,882] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] (Re-)joining group (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:605)
[2025-01-03 17:20:16,899] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] (Re-)joining group (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:605)
[2025-01-03 17:20:16,900] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Successfully joined group with generation Generation{generationId=7, memberId='connect-192.168.1.5:8083-a793d6a5-8bf8-4ff7-a6a4-96c7f338e6b8', protocol='sessioned'} (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:666)
[2025-01-03 17:20:16,902] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Successfully synced group in generation Generation{generationId=7, memberId='connect-192.168.1.5:8083-a793d6a5-8bf8-4ff7-a6a4-96c7f338e6b8', protocol='sessioned'} (org.apache.kafka.connect.runtime.distributed.WorkerCoordinator:843)
[2025-01-03 17:20:16,902] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Joined group at generation 7 with protocol version 2 and got assignment: Assignment{error=0, leader='connect-192.168.1.5:8083-a793d6a5-8bf8-4ff7-a6a4-96c7f338e6b8', leaderUrl='http://192.168.1.5:8083/', offset=10, connectorIds=[mongodb-sink, mariadb-connector], taskIds=[mongodb-sink-0, mariadb-connector-0], revokedConnectorIds=[], revokedTaskIds=[], delay=0} with rebalance delay: 0 (org.apache.kafka.connect.runtime.distributed.DistributedHerder:2648)
[2025-01-03 17:20:16,903] WARN [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Catching up to assignment's config offset. (org.apache.kafka.connect.runtime.distributed.DistributedHerder:1811)
[2025-01-03 17:20:16,903] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Current config state offset 7 is behind group assignment 10, reading to end of config log (org.apache.kafka.connect.runtime.distributed.DistributedHerder:1884)
[2025-01-03 17:20:16,903] INFO [Consumer clientId=connect-cluster-configs, groupId=connect-cluster] Disconnecting from node 0 due to request timeout. (org.apache.kafka.clients.NetworkClient:871)
[2025-01-03 17:20:16,903] INFO [Consumer clientId=connect-cluster-configs, groupId=connect-cluster] Cancelled in-flight FETCH request with correlation id 23033 due to node 0 being disconnected (elapsed time since creation: 925403ms, elapsed time since send: 925403ms, throttle time: 0ms, request timeout: 30000ms) (org.apache.kafka.clients.NetworkClient:364)
[2025-01-03 17:20:16,905] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Finished reading to end of log and updated config snapshot, new config log offset: 10 (org.apache.kafka.connect.runtime.distributed.DistributedHerder:1911)
[2025-01-03 17:20:16,905] INFO [Consumer clientId=connect-cluster-configs, groupId=connect-cluster] Error sending fetch request (sessionId=723236997, epoch=22945) to node 0: (org.apache.kafka.clients.FetchSessionHandler:618)
org.apache.kafka.common.errors.DisconnectException
[2025-01-03 17:20:16,905] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Starting connectors and tasks using config offset 10 (org.apache.kafka.connect.runtime.distributed.DistributedHerder:1979)
[2025-01-03 17:20:16,905] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Finished starting connectors and tasks (org.apache.kafka.connect.runtime.distributed.DistributedHerder:2008)
[2025-01-03 17:20:17,021] INFO [Worker clientId=connect-192.168.1.5:8083, groupId=connect-cluster] Session key updated (org.apache.kafka.connect.runtime.distributed.DistributedHerder:2510)
[2025-01-03 17:20:25,548] INFO [mariadb-connector|task-0] Stopped reading binlog after 3 events, last recorded offset: {ts_sec=1735940249, file=mariadb-bin.000001, pos=88942, server_id=1, event=1} (io.debezium.connector.binlog.BinlogStreamingChangeEventSource:1218)
[2025-01-03 17:20:25,548] INFO [mariadb-connector|task-0] Creating thread debezium-mysqlconnector-kafka-binlog-client (io.debezium.util.Threads:287)
[2025-01-03 17:20:25,586] INFO [mariadb-connector|task-0] Connected to binlog at localhost:3306, starting at BinlogOffsetContext{sourceInfoSchema=Schema{io.debezium.connector.mysql.Source:STRUCT}, sourceInfo=BinlogSourceInfo{currentGtid='null', currentBinlogFilename='mariadb-bin.000001', currentBinlogPosition=88901, currentRowNumber=0, serverId=1, sourceTime=2025-01-03T21:37:29Z, threadId=-1, currentQuery='null', tableIds=[leafy_factory.product_cost], databaseName='leafy_factory'}, snapshotCompleted=true, transactionContext=TransactionContext [currentTransactionId=null, perTableEventCount={}, totalEventCount=0], restartGtidSet='null', currentGtidSet='null', restartBinlogFilename='mariadb-bin.000001', restartBinlogPosition=88942, restartRowsToSkip=0, restartEventsToSkip=1, currentEventLengthInBytes=41, inTransaction=false, transactionId='null', incrementalSnapshotContext=IncrementalSnapshotContext [windowOpened=false, chunkEndPosition=null, dataCollectionsToSnapshot=[], lastEventKeySent=null, maximumKey=null]} (io.debezium.connector.binlog.BinlogStreamingChangeEventSource:1232)
[2025-01-03 17:20:28,593] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:34, serverValue:1324294}] to ist-shared-shard-00-01.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:20:29,191] INFO [mongodb-sink|task-0] Exception in monitor thread while connecting to server ist-shared-shard-00-00.n0kts.mongodb.net:27017 (org.mongodb.driver.cluster:76)
com.mongodb.MongoSocketReadTimeoutException: Timeout while receiving message
	at com.mongodb.internal.connection.InternalStreamConnection.translateReadException(InternalStreamConnection.java:696)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:574)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveCommandMessageResponse(InternalStreamConnection.java:410)
	at com.mongodb.internal.connection.InternalStreamConnection.receive(InternalStreamConnection.java:369)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.lookupServerDescription(DefaultServerMonitor.java:221)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.run(DefaultServerMonitor.java:157)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:278)
	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:304)
	at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:346)
	at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:796)
	at java.base/java.net.Socket$SocketInputStream.implRead(Socket.java:1116)
	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:1103)
	at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:489)
	at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:483)
	at java.base/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:70)
	at java.base/sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1462)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1068)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:110)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:135)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveResponseBuffers(InternalStreamConnection.java:713)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:571)
	... 5 more
[2025-01-03 17:20:29,822] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:35, serverValue:1173219}] to ist-shared-shard-00-00.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:20:29,823] INFO [mongodb-sink|task-0] Monitor thread successfully connected to server with description ServerDescription{address=ist-shared-shard-00-00.n0kts.mongodb.net:27017, type=REPLICA_SET_SECONDARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=25, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=472500709, setName='atlas-133bp7-shard-0', canonicalAddress=ist-shared-shard-00-00.n0kts.mongodb.net:27017, hosts=[ist-shared-shard-00-02.n0kts.mongodb.net:27017, ist-shared-shard-00-01.n0kts.mongodb.net:27017, ist-shared-shard-00-00.n0kts.mongodb.net:27017], passives=[], arbiters=[], primary='ist-shared-shard-00-01.n0kts.mongodb.net:27017', tagSet=TagSet{[Tag{name='availabilityZone', value='cac1-az1'}, Tag{name='diskState', value='READY'}, Tag{name='nodeType', value='ELECTABLE'}, Tag{name='provider', value='AWS'}, Tag{name='region', value='CA_CENTRAL_1'}, Tag{name='workloadType', value='OPERATIONAL'}]}, electionId=null, setVersion=71, topologyVersion=TopologyVersion{processId=675b24d1d1431a587be694a8, counter=19}, lastWriteDate=Fri Jan 03 17:20:29 CST 2025, lastUpdateTimeNanos=66537468300958} (org.mongodb.driver.cluster:71)
[2025-01-03 17:20:30,835] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:36, serverValue:1173221}] to ist-shared-shard-00-00.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:20:31,034] INFO [mongodb-sink|task-0] Exception in monitor thread while connecting to server ist-shared-shard-00-02.n0kts.mongodb.net:27017 (org.mongodb.driver.cluster:76)
com.mongodb.MongoSocketReadTimeoutException: Timeout while receiving message
	at com.mongodb.internal.connection.InternalStreamConnection.translateReadException(InternalStreamConnection.java:696)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:574)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveCommandMessageResponse(InternalStreamConnection.java:410)
	at com.mongodb.internal.connection.InternalStreamConnection.receive(InternalStreamConnection.java:369)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.lookupServerDescription(DefaultServerMonitor.java:221)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.run(DefaultServerMonitor.java:157)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:278)
	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:304)
	at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:346)
	at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:796)
	at java.base/java.net.Socket$SocketInputStream.implRead(Socket.java:1116)
	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:1103)
	at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:489)
	at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:483)
	at java.base/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:70)
	at java.base/sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1462)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1068)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:110)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:135)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveResponseBuffers(InternalStreamConnection.java:713)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:571)
	... 5 more
[2025-01-03 17:20:31,524] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:37, serverValue:714889}] to ist-shared-shard-00-02.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:20:31,526] INFO [mongodb-sink|task-0] Monitor thread successfully connected to server with description ServerDescription{address=ist-shared-shard-00-02.n0kts.mongodb.net:27017, type=REPLICA_SET_SECONDARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=25, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=391716875, setName='atlas-133bp7-shard-0', canonicalAddress=ist-shared-shard-00-02.n0kts.mongodb.net:27017, hosts=[ist-shared-shard-00-02.n0kts.mongodb.net:27017, ist-shared-shard-00-01.n0kts.mongodb.net:27017, ist-shared-shard-00-00.n0kts.mongodb.net:27017], passives=[], arbiters=[], primary='ist-shared-shard-00-01.n0kts.mongodb.net:27017', tagSet=TagSet{[Tag{name='availabilityZone', value='cac1-az4'}, Tag{name='diskState', value='READY'}, Tag{name='nodeType', value='ELECTABLE'}, Tag{name='provider', value='AWS'}, Tag{name='region', value='CA_CENTRAL_1'}, Tag{name='workloadType', value='OPERATIONAL'}]}, electionId=null, setVersion=71, topologyVersion=TopologyVersion{processId=67694a2575e55e11df78dcd1, counter=14}, lastWriteDate=Fri Jan 03 17:20:31 CST 2025, lastUpdateTimeNanos=66539170737000} (org.mongodb.driver.cluster:71)
[2025-01-03 17:20:32,026] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:38, serverValue:714890}] to ist-shared-shard-00-02.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:20:36,095] INFO [mongodb-sink|task-0] Exception in monitor thread while connecting to server ist-shared-shard-00-01.n0kts.mongodb.net:27017 (org.mongodb.driver.cluster:76)
com.mongodb.MongoSocketReadTimeoutException: Timeout while receiving message
	at com.mongodb.internal.connection.InternalStreamConnection.translateReadException(InternalStreamConnection.java:696)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:574)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveCommandMessageResponse(InternalStreamConnection.java:410)
	at com.mongodb.internal.connection.InternalStreamConnection.receive(InternalStreamConnection.java:369)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.lookupServerDescription(DefaultServerMonitor.java:221)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.run(DefaultServerMonitor.java:157)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:278)
	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:304)
	at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:346)
	at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:796)
	at java.base/java.net.Socket$SocketInputStream.implRead(Socket.java:1116)
	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:1103)
	at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:489)
	at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:483)
	at java.base/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:70)
	at java.base/sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1462)
	at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1068)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:110)
	at com.mongodb.internal.connection.SocketStream.read(SocketStream.java:135)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveResponseBuffers(InternalStreamConnection.java:713)
	at com.mongodb.internal.connection.InternalStreamConnection.receiveMessageWithAdditionalTimeout(InternalStreamConnection.java:571)
	... 5 more
[2025-01-03 17:20:36,507] INFO [mongodb-sink|task-0] Opened connection [connectionId{localValue:39, serverValue:1324313}] to ist-shared-shard-00-01.n0kts.mongodb.net:27017 (org.mongodb.driver.connection:71)
[2025-01-03 17:20:36,508] INFO [mongodb-sink|task-0] Monitor thread successfully connected to server with description ServerDescription{address=ist-shared-shard-00-01.n0kts.mongodb.net:27017, type=REPLICA_SET_PRIMARY, state=CONNECTED, ok=true, minWireVersion=0, maxWireVersion=25, maxDocumentSize=16777216, logicalSessionTimeoutMinutes=30, roundTripTimeNanos=304405916, setName='atlas-133bp7-shard-0', canonicalAddress=ist-shared-shard-00-01.n0kts.mongodb.net:27017, hosts=[ist-shared-shard-00-02.n0kts.mongodb.net:27017, ist-shared-shard-00-01.n0kts.mongodb.net:27017, ist-shared-shard-00-00.n0kts.mongodb.net:27017], passives=[], arbiters=[], primary='ist-shared-shard-00-01.n0kts.mongodb.net:27017', tagSet=TagSet{[Tag{name='availabilityZone', value='cac1-az2'}, Tag{name='diskState', value='READY'}, Tag{name='nodeType', value='ELECTABLE'}, Tag{name='provider', value='AWS'}, Tag{name='region', value='CA_CENTRAL_1'}, Tag{name='workloadType', value='OPERATIONAL'}]}, electionId=7fffffff000000000000010a, setVersion=71, topologyVersion=TopologyVersion{processId=67694b4e6118cd58a4791747, counter=41}, lastWriteDate=Fri Jan 03 17:20:36 CST 2025, lastUpdateTimeNanos=66544153449083} (org.mongodb.driver.cluster:71)
[2025-01-03 17:20:36,509] INFO [mongodb-sink|task-0] Discovered replica set primary ist-shared-shard-00-01.n0kts.mongodb.net:27017 (org.mongodb.driver.cluster:71)
